### PR TITLE
Resolve v0.18.1 MPI/HDF5 issue

### PR DESCRIPTION
**Pull request by rbberger**
## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file

### CODE REVIEW

**Comment by ktsai7** [comment]
@rbberger can we wait on this for a little bit? After I merged in the change to update spack to v0.18.1 for the project spack, it's best to just leave both `SINGULARITY_EOS_TEMP_SPACK_VERSION` and `SINGULARITY_EOS_TEMP_SPACK_DIR` empty and it should use the project spack version and directory automatically.

**Comment by rbberger** [comment]
Sure. This one was just to show that the one remaining issue you mentioned is resolved if you bump the OpenMPI version.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -9,9 +9,9 @@ stages:
 variables:
   GIT_SUBMODULE_STRATEGY: recursive
   SINGULARITY_EOS_GCC_VERSION: "9.3.0"
-  SINGULARITY_EOS_OPENMPI_VERSION: "4.0.5"
-  SINGULARITY_EOS_TEMP_SPACK_VERSION: "v0.16.3"
-  SINGULARITY_EOS_TEMP_SPACK_DIR: "/projects/xcap/oss/deployment/spack-v0.16.3-3"
+  SINGULARITY_EOS_OPENMPI_VERSION: "4.1.3"
+  SINGULARITY_EOS_TEMP_SPACK_VERSION: "v0.18.1"
+  SINGULARITY_EOS_TEMP_SPACK_DIR: "/projects/xcap/oss/deployment/CI/spack-v0.18.1-0"
```
Comment (ktsai7) [inline .gitlab-ci.yml:14]: @rbberger try leaving these two just `""`

Comment (rbberger) [reply .gitlab-ci.yml]: @ktsai7 Seems to work.

**Comment by Yurlungur** [comment]
What is the status of this? Is it still needed? And is it still on hold? If it's on hold, let's change to WIP for now.

**Comment by ktsai7** [comment]
@Yurlungur this should be ready to go

**Review by Yurlungur — APPROVED** [review]

**Pull request merged**